screens/Reader.js

```diff
@@ -77,9 +81,33 @@ class Reader extends React.Component {
   }
 
   componentDidMount() {
-    this.loadShabad();
+      this.loadShabad();
+      this.setPosition();
+
+  }
+
+  setPosition(){
+    const startBaniList=JSON.parse(this.props.startBani);
+      let progress=0;
+      if(startBaniList.length>0){
+        for(const bani of startBaniList){
+          if(bani.id==this.currentBani.id){
+            if(bani.progress>0 && bani.progress<1){
+            progress=bani.progress
+            }
+          }
+        }
+      }
+      this.currentBani.progress=progress
   }
 
+  handleBackPress(data){
+    this.webView.postMessage(JSON.stringify({ Back: true }))
+    this.props.navigation.goBack();
+    
+  }
+
```

Comment:
Please fix indentation issues in this file. At some places semicolon is also missing.

Reply:
Fixed